### DISCUSSION TITLE
@formily/vue + vue3 RecursionField自增列表中包含输入框时form values 异常

### DISCUSSION THREAD

**Discussion by CatalystDP** [alibaba/formily#2618] [Q&A 2.x] [answered]
vue3 下RecursionField 自增列表中input，在输入后触发blur事件后，单个input 组件的值会覆盖掉自增列表所在字段的值，
codesandbox 如下：
https://codesandbox.io/s/formily-vue3-ozlhn

vue2 则没有这个问题，对应codesandbox 
https://codesandbox.io/s/formily-vue2-r6lbw

请大佬帮忙看看是不是哪里使用有问题？

**Comment by MisicDemone** [comment, marked as answer]
禁用掉 input 组件change事件的冒泡即可
``` js
const Input = defineComponent({
  name: "Input",
  props: ['value'],
  setup(props, {  emit }) {
    return () => {
      return h("input", {
        value: props.value,
        type: "text",
        onInput: (ev) => {
          emit('change', ev.target.value)
        },
        onChange: (ev) => {
          ev.stopPropagation()
        },
      });
    };
  },
})
```

**Reply by CatalystDP** [reply]
可以了，但是是什么原因呢?

**Reply by MisicDemone** [reply]
> 可以了，但是是什么原因呢?

因为formily是通过change事件来进行视图和模型的关联的，input组件自身的change事件也会产生影响

**Comment by XQ0118** [comment]
使用 antdv 组件库的 AInput 组件也会有这个问题，请大佬帮忙看下？

**Reply by CatalystDP** [reply]
在AInput 外包一层div 然后监听onChange事件阻止冒泡试试，
